src/draw/sw/lv_draw_sw.c

```diff
@@ -226,6 +226,26 @@ void lv_draw_sw_rgb565_swap(void * buf, uint32_t buf_size_px)
 
 }
 
+void lv_draw_sw_i1_invert(void * buf, uint32_t buf_size)
+{
+    if(buf == NULL) return;
+
+    uint32_t * word_buf = (uint32_t *)buf;
```

Comment:
As ```buf``` is an array of `uint8_t`, I think that there is no alignment guarantee.
Performing 16 or 32 bit wide operations on non aligned addresses does not
work reliable on non x86 platforms.

You may use the implementation of [memset](https://github.com/eblot/newlib/blob/master/newlib/libc/string/memset.c) as an example how this could be implemented.

Reply:
True, did not consider this. Added a version inspired by memset :^)